### PR TITLE
feat: Add blogpost about generic react context

### PR DESCRIPTION
**Pull request by orjandesmet**
### Thank you for your contribution! Please go through this small checklist to prepare your pull request to be reviewed as soon as possible :)
* [x] All authors were added to `_data/authors.yml` with a picture and an up-to-date bio
* [x] The blog post image is of ratio 500x293
  * Feel free to leave a comment if you don't know how to resize your image and someone will gladly help you out
* [x] The [**one sentence per line** method](https://raw.githubusercontent.com/brandon-rhodes/blog/master/texts/brandon/2012/one-sentence-per-line.rst) has been adopted
* [x] A spell check has been done
* [x] All links open in a new tab
  * Example: `[Amazing tech blog](https://ordina-jworks.github.io){:target="_blank" rel="noopener noreferrer"}`
* [x] The post's content properly resizes according to the window's size
  * Tip: Just shorten your window's horizontal length to easily test this
  * To have images properly resize use `class="image fit"`
  * To have YouTube videos properly resize, wrap the iframe into a `<div class="responsive-video">`
* [x] If a table of content is: check whether all links are working correctly

### Any specific remarks for the reviewers or anything you need help with?
\<enter text here>

### Checklist for the reviewers before merging
* [ ] Publishing date of the post has been correctly set
  * This can be done by renaming the filename or by setting the date in the header section of the post

### CODE REVIEW

File: _posts/2020-12-18-react-generic-context.md [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+These components are considered building blocks and can consist themselves of smaller components.
+These components have properties, being passed by their parents and can these properties can influence the rendered view and logic of the components.
+
+Those with a bit more React experience, will know that in a larger tree of components, it’s a nuisance to get a property from a component near the root of the tree to a component near the leafs of the tree by simple property binding.
+It also muddies intermediary components with irrelevant code.
+
+Fortunately React includes **Context** [Docs](https://reactjs.org/docs/context.html){:target="_blank" rel="noopener noreferrer"}.
+It provides a way to share value like preferences/themes and authentication data without having to explicitly set those values in each component’s properties.
+
+The developer basically creates a `Context` variable which is provided by a near root component using a `Provider`, and can be consumed by other components  using either a `Consumer` or a `useContext` hook.
```
Comment (sjbuysse) [inline _posts/2020-12-18-react-generic-context.md:27]: * What is a root component? Do you mean container component? 

* using a `Provider` component
* eithar a `Consumer` component or a `useContext` hook

File: _posts/2020-12-18-react-generic-context.md [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+## A background
+<img class="image fit center" alt="A picture of the Windows XP background" src="/img/react-generic-context/a-background.jpg">
+
+I have been mostly working with Angular for the past 4 years.
+I am, however, a firm believer that no one framework or library is better than all others and I love a sporadic challenging dive into the unknown.
+
+Those who know a little bit about React will know that, like Angular, Vue and others, it’s a JavaScript library to create web applications by means of components.
+These components are considered building blocks and can consist themselves of smaller components.
+These components have properties, being passed by their parents and can these properties can influence the rendered view and logic of the components.
```
Comment (sjbuysse) [inline _posts/2020-12-18-react-generic-context.md:19]: ... by their parents and these properties... (there is a can too much)

File: _posts/2020-12-18-react-generic-context.md [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+```typescript
+// authentication-oidc/factory.ts
+import { createAuthService } from '@our-scope/authentication-core';
+import { OidcProps } from './domain/oidc-props';
+import { OidcAuthService } from './services/oidc-auth.service';
+
+export function createOidcAuthService(oidcProps: OidcProps) {
+  const oidcService = new OidcAuthService(oidcProps);
+  return createAuthService(oidcService);
```
Comment (sjbuysse) [inline _posts/2020-12-18-react-generic-context.md:167]: Should this createAuthService get a type now? Or is that inferred somehow from the oidcService (if that's the case, it's not superclear)?

Comment (orjandesmet) [reply _posts/2020-12-18-react-generic-context.md:167]: It's inferred. I'll mention it

**Comment by yannickdeturck** [comment]
Looks like the post is good to go, @orjandesmet, @sjbuysse ?

**Review by sjbuysse — APPROVED** [review]

**Pull request merged**